app/Http/Requests/ProfilesApiRequest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Profile;
+use App\Rules\AllowedProfileDataType;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+class ProfilesApiRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules()
+    {
+        return [
+            'person' => 'sometimes|string',
+            'search' => 'sometimes|string',
+            'search_names' => 'sometimes|string',
+            'info_contains' => 'sometimes|string',
+            'from_school' => 'sometimes|string',
+            'tag' => 'sometimes|string',
+            'public' => 'sometimes|boolean',
```

Comment:
For the public parameter, I did enter true and false, and still didn't pass the validation. Perhaps, although it might be implicit, the message should say allowed values are 1 for true and 0 for false.